### DISCUSSION TITLE
Export/import database schema

### DISCUSSION THREAD

**Discussion by Magnuti** [pocketbase/pocketbase#2358] [Q&A] [answered]
Is it possible to export the database schema?

I would like to create and manage collections in the web UI and then import the created database schema/structure into another instance of PocketBase. This way I can keep the DB structure in source control and simply spin up the database in Docker.

**Comment by ganigeorgiev** [comment, marked as answer]
This is already supported.

You can export the collections configuration as JSON from the "_Admin UI > Settings > Export collections_" and reimport it from "_Admin UI > Settings > Import collections_".

If you want a programmatical approach, I recommend using [Migrations](https://pocketbase.io/docs/migrations/). Both Go and JS (_not documented very well at the moment_) migrations are supported.

If you use the prebuilt executable, automigrate is enabled by default, so any collections change from the Admin UI should generate a migration file in the `pb_migrations` directory. This directory is safe to be committed in your repo and all unapplied migrations will be automatically applied on `serve`.
